### PR TITLE
IOU per sample calculation

### PR DESCRIPTION
**Pull request by JonasWurst**
## What has changed and why?

Added a methoid for calculating the Intersection over Union (IoU) for object detection evaluation. Will be used by the object detction evalaution.

## How has it been tested?

Implemented unit tests to verify the correctness of the IoU calculations under various scenarios, including empty inputs, perfect overlap, and known partial overlaps.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Summary by CodeRabbit

* **New Features**
  * Added Intersection over Union (IoU) metric computation for evaluating object detection predictions against ground-truth bounding boxes, handling empty and degenerate cases.

* **Tests**
  * Added comprehensive test suite for the IoU metric covering empty inputs, perfect overlap, no overlap, edge-touching, partial overlap, degenerate (zero-area) boxes, negative coordinates, output shape, and symmetry.

### CODE REVIEW

File: lightly_studio/src/lightly_studio/evaluation/object_detection_metric.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"""Object detection evaluation metric primitives."""
+
+from __future__ import annotations
+
+import numpy as np
+
+
+def compute_iou_matrix(pred_corners: np.ndarray, gt_corners: np.ndarray) -> np.ndarray:
+    """Compute pairwise IoU from corner arrays.
```
Comment (IgorSusmelj) [inline lightly_studio/src/lightly_studio/evaluation/object_detection_metric.py:13]: It's not relevant for the computation itself but please mention if we expect integers (pixel coords) or floats (between 0,1) as input

File: lightly_studio/src/lightly_studio/evaluation/object_detection_metric.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    Args:
+        pred_corners: Array of shape (P, 4) with [x1, y1, x2, y2] per prediction.
+        gt_corners: Array of shape (G, 4) with [x1, y1, x2, y2] per ground truth.
+
+    Returns:
+        IoU matrix of shape (P, G), or an empty array if either input is empty.
+    """
+    if pred_corners.size == 0 or gt_corners.size == 0:
```
Comment (IgorSusmelj) [inline lightly_studio/src/lightly_studio/evaluation/object_detection_metric.py:22]: We don't check for negative numbers (which is good). Sometimes boxes might be out of the image and in certain edge cases you want to keep it that way and not clamp the boxes. We don't have a test for this. Does it still work?

**Review by IgorSusmelj — CHANGES_REQUESTED** [review]
Please make the docstring more clear and add a test for negative coords (e.g. a box with coords [-3,-1,20,20]. Otherwise looks good to me.

**Review by IgorSusmelj — APPROVED** [review]
LGTM
